miner/worker.go

```diff
@@ -88,17 +89,19 @@ type environment struct {
 	tcount    int            // tx count in cycle
 	gasPool   *core.GasPool  // available gas used to pack transactions
 
-	header   *types.Header
-	txs      []*types.Transaction
-	receipts []*types.Receipt
+	header           *types.Header
+	txs              []*types.Transaction
+	receipts         []*types.Receipt
+	executionResults []*types.ExecutionResult
 }
 
 // task contains all information for consensus engine sealing and result submitting.
 type task struct {
-	receipts  []*types.Receipt
-	state     *state.StateDB
-	block     *types.Block
-	createdAt time.Time
+	receipts         []*types.Receipt
+	executionResults []*types.ExecutionResult
```

Comment:
haicheng's feeback is not really resolved

Reply:
Don't need to changed here.

Reply:
Probably not enough as I explained above

Reply:
I think @mask-pp 's approach is doable, and tend to agree with him: I also prefer building blockResult just before committing a block. account_proof storage_proof is corresonding to a block